samples/algorithms/ThreeQubitRepetitionCode.qs

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.Quantum.Samples {
+    open Microsoft.Quantum.Math;
+    open Microsoft.Quantum.Diagnostics;
+
+    @EntryPoint()
+    operation ThreeQubitRepetitionCode() : (Bool, Int) {
+        // A qubit register that will be used for encoding.
+        use encodedRegister = Qubit[3];
+
+        // Initialize the first qubit in the register to a |-〉 state.
+        H(encodedRegister[0]);
+        Z(encodedRegister[0]);
+
+        // Apply several unitary operations to the encoded qubits
+        // performing bit flip detection and correction between each application.
+        mutable bitFlipCount = 0;
+        within {
+            // The 3 qubit register is used as a repetition code.
+            Encode(encodedRegister);
+        } apply {
+            let iterations = 5;
+            for _ in 1..iterations {
+                // Apply a unitary operatyion to the encoded register that shuold
+                // effectively perform an identity operation but may be noisy
+                // on the quantum hardware and introduce errors.
+                ApplyRotationalIdentity(encodedRegister);
+
+                // Measure the bit flip error syndrome, revert the bit flip if needed,
+                // and increase the count if a bit flip occurred.
+                let (parity01, parity12) = MeasureBitFlipSyndrome(encodedRegister);
+                let bitFlipReverted = RevertBitFlip(encodedRegister, parity01, parity12);
+                if (bitFlipReverted) {
+                    set bitFlipCount += 1;
+                }
+            }
+        }
+
+        // Transform the qubit to the |1〉 state and measure it in the computational basis.
+        H(encodedRegister[0]);
+        let result = MResetZ(encodedRegister[0]) == One;
+        ResetAll(encodedRegister);
```

Comment:
Nitpick: not really important since it's right at the end of execution, but the first qubit here was already reset by the line above, so this could be:
```suggestion
        ResetAll(encodedRegister[1...]);
```

Reply:
Updated and added comment.